Core/TextureReplacer.cpp

```diff
@@ -1181,31 +1053,13 @@ void ReplacedTexture::PrepareData(int level) {
 		png_image png = {};
 		png.version = PNG_IMAGE_VERSION;
 
-		// Needs to survive for a little while, used for zip only.
 		std::string pngdata;
-		if (fp) {
-			if (!png_image_begin_read_from_stdio(&png, fp)) {
-				ERROR_LOG(G3D, "Could not load texture replacement info: %s - %s", info.file.c_str(), png.message);
-				cleanup();
-				return;
-			}
-		} else if (zf) {
-			zip_uint64_t zsize = ZipFileSize(info.zinfo->z, info.zi);
-			if (zsize != INVALID_ZIP_SIZE)
-				pngdata.resize(zsize);
-			if (!pngdata.empty()) {
-				pngdata.resize(zip_fread(zf, &pngdata[0], pngdata.size()));
-			}
-			// Try to unlock early to prevent blocking threads.
-			zguard.unlock();
-
-			if (!png_image_begin_read_from_memory(&png, &pngdata[0], pngdata.size())) {
-				ERROR_LOG(G3D, "Could not load texture replacement info: %s - %s (zip)", info.file.c_str(), png.message);
-				cleanup();
-				return;
-			}
-		} else {
-			_assert_(false);
+		pngdata.resize(fileSize);
+		vfs_->Read(openFile, &pngdata[0], fileSize);
```

Comment:
I still think it'd be more correct here to use `pngdata.resize(vfs_->Read(openFile, &pngdata[0], fileSize));` since the call to `png_image_begin_read_from_memory()` is more likely to error properly if the read is truncated somehow.

-[Unknown]

Reply:
Hm, fair enough. Though it's a buffered read, not a syscall read, so it should only be truncated if the actual file was truncated in the meantime, which isn't too likely. Still, I'll revert these.